From e3eb56ada370c18c1539e29b929139598a366611 Mon Sep 17 00:00:00 2001
From: Mikolaj Izdebski <mizdebsk@redhat.com>
Date: Wed, 13 Nov 2013 14:32:23 +0100
Subject: [PATCH 3/3] Update to Sisu 0.1.0 and Guice 3.1.6

Sisu depends on Guice, but dependency scope changed from "compile" to
"provided" in Sisu 0.1.0.  As a Sisu user, Maven needs to have runtime
dependency on Guice.
---
 apache-maven/pom.xml          | 12 ++++++++++++
 maven-aether-provider/pom.xml |  6 +++++-
 maven-compat/pom.xml          | 12 ++++++++++++
 maven-core/pom.xml            | 11 +++++++++++
 maven-embedder/pom.xml        | 11 +++++++++++
 maven-model-builder/pom.xml   | 12 ++++++++++++
 maven-plugin-api/pom.xml      | 12 ++++++++++++
 pom.xml                       |  4 ++--
 8 files changed, 77 insertions(+), 3 deletions(-)

diff --git a/apache-maven/pom.xml b/apache-maven/pom.xml
index 2923435..16393cb 100644
--- a/apache-maven/pom.xml
+++ b/apache-maven/pom.xml
@@ -57,6 +57,18 @@
       <groupId>org.eclipse.sisu</groupId>
       <artifactId>org.eclipse.sisu.plexus</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.sonatype.sisu</groupId>
+      <artifactId>sisu-guice</artifactId>
+      <classifier>no_aop</classifier>
+      <scope>runtime</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>aopalliance</groupId>
+          <artifactId>aopalliance</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
     <!-- CLI -->
     <dependency>
       <groupId>commons-cli</groupId>
diff --git a/maven-aether-provider/pom.xml b/maven-aether-provider/pom.xml
index 801f9fb..4fbca7d 100644
--- a/maven-aether-provider/pom.xml
+++ b/maven-aether-provider/pom.xml
@@ -80,7 +80,6 @@ under the License.
       <groupId>org.sonatype.sisu</groupId>
       <artifactId>sisu-guice</artifactId>
       <classifier>no_aop</classifier>
-      <optional>true</optional>
       <exclusions>
         <exclusion>
           <groupId>aopalliance</groupId>
@@ -88,6 +87,11 @@ under the License.
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>javax.inject</groupId>
+      <artifactId>javax.inject</artifactId>
+      <version>1</version>
+    </dependency>
     <!-- Testing -->
     <dependency>
       <groupId>org.eclipse.aether</groupId>
diff --git a/maven-compat/pom.xml b/maven-compat/pom.xml
index 348ce29..7884159 100644
--- a/maven-compat/pom.xml
+++ b/maven-compat/pom.xml
@@ -64,6 +64,18 @@
       <artifactId>org.eclipse.sisu.plexus</artifactId>
     </dependency>
     <dependency>
+      <groupId>org.sonatype.sisu</groupId>
+      <artifactId>sisu-guice</artifactId>
+      <classifier>no_aop</classifier>
+      <scope>runtime</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>aopalliance</groupId>
+          <artifactId>aopalliance</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
       <groupId>org.codehaus.plexus</groupId>
       <artifactId>plexus-component-annotations</artifactId>
     </dependency>
diff --git a/maven-core/pom.xml b/maven-core/pom.xml
index fae2904..dc788d2 100644
--- a/maven-core/pom.xml
+++ b/maven-core/pom.xml
@@ -82,6 +82,17 @@
       <artifactId>org.eclipse.sisu.plexus</artifactId>
     </dependency>
     <dependency>
+      <groupId>org.sonatype.sisu</groupId>
+      <artifactId>sisu-guice</artifactId>
+      <classifier>no_aop</classifier>
+      <exclusions>
+        <exclusion>
+          <groupId>aopalliance</groupId>
+          <artifactId>aopalliance</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
       <groupId>org.codehaus.plexus</groupId>
       <artifactId>plexus-interpolation</artifactId>
     </dependency>
diff --git a/maven-embedder/pom.xml b/maven-embedder/pom.xml
index 2da52a9..f246fe1 100644
--- a/maven-embedder/pom.xml
+++ b/maven-embedder/pom.xml
@@ -65,6 +65,17 @@
       <artifactId>org.eclipse.sisu.plexus</artifactId>
     </dependency>
     <dependency>
+      <groupId>org.sonatype.sisu</groupId>
+      <artifactId>sisu-guice</artifactId>
+      <classifier>no_aop</classifier>
+      <exclusions>
+        <exclusion>
+          <groupId>aopalliance</groupId>
+          <artifactId>aopalliance</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
       <groupId>org.codehaus.plexus</groupId>
       <artifactId>plexus-component-annotations</artifactId>
     </dependency>
diff --git a/maven-model-builder/pom.xml b/maven-model-builder/pom.xml
index 81cd9d9..c1ca303 100644
--- a/maven-model-builder/pom.xml
+++ b/maven-model-builder/pom.xml
@@ -54,6 +54,18 @@
       <scope>test</scope>
     </dependency>
     <dependency>
+      <groupId>org.sonatype.sisu</groupId>
+      <artifactId>sisu-guice</artifactId>
+      <classifier>no_aop</classifier>
+      <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>aopalliance</groupId>
+          <artifactId>aopalliance</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
       <groupId>xmlunit</groupId>
       <artifactId>xmlunit</artifactId>
       <version>1.3</version>
diff --git a/maven-plugin-api/pom.xml b/maven-plugin-api/pom.xml
index 1019ae4..e018cc0 100644
--- a/maven-plugin-api/pom.xml
+++ b/maven-plugin-api/pom.xml
@@ -58,6 +58,18 @@ under the License.
       <groupId>org.eclipse.sisu</groupId>
       <artifactId>org.eclipse.sisu.plexus</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.sonatype.sisu</groupId>
+      <artifactId>sisu-guice</artifactId>
+      <classifier>no_aop</classifier>
+      <scope>runtime</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>aopalliance</groupId>
+          <artifactId>aopalliance</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
   </dependencies>
 
   <build>
diff --git a/pom.xml b/pom.xml
index 9d858fa..3787a42 100644
--- a/pom.xml
+++ b/pom.xml
@@ -52,8 +52,8 @@
     <plexusUtilsVersion>3.0.17</plexusUtilsVersion>
     <!-- Latest version of Guava that works with Sisu -->
     <guavaVersion>14.0.1</guavaVersion>
-    <guiceVersion>3.1.3</guiceVersion>
-    <sisuInjectVersion>0.0.0.M5</sisuInjectVersion>
+    <guiceVersion>3.1.6</guiceVersion>
+    <sisuInjectVersion>0.1.0</sisuInjectVersion>
     <wagonVersion>2.6</wagonVersion>
     <securityDispatcherVersion>1.3</securityDispatcherVersion>
     <cipherVersion>1.7</cipherVersion>
-- 
1.9.3

